elftools/ehabi/decoder.py

```diff
@@ -0,0 +1,271 @@
+                Integer array, raw data of bytecode.
+
+            auto_decode:
+                bool, whether bytecode should be decoded during construction.
+                (default : true)
+    """
+
+    def __init__(self, bytecode_array, auto_decode=True):
+        self._bytecode_array = bytecode_array
+        self._index = None
+        self.mnemonic_array = None
+        if auto_decode:
+            self.decode()
+
+    def decode(self):
+        """ Decode bytecode array, put result into mnemonic_array.
+        """
+        self._index = 0
+        self.mnemonic_array = []
+        while self._index < len(self._bytecode_array):
+            for mask, value, _Decode_handler in self.ring:
+                if (self._bytecode_array[self._index] & mask) == value:
+                    start_idx = self._index
+                    mnemonic = _Decode_handler(self)
+                    end_idx = self._index
+                    self.mnemonic_array.append(MnemonicItem(self._bytecode_array[start_idx: end_idx], mnemonic))
+                    break
+
+    def _Decode_00xxxxxx(self):
+        #   SW.startLine() << format("0x%02X      ; vsp = vsp + %u\n", Opcode,
+        #                            ((Opcode & 0x3f) << 2) + 4);
+        opcode = self._bytecode_array[self._index]
+        self._index += 1
+        return 'vsp = vsp + %u' % (((opcode & 0x3f) << 2) + 4)
+
+    def _Decode_01xxxxxx(self):
+        # SW.startLine() << format("0x%02X      ; vsp = vsp - %u\n", Opcode,
+        #                          ((Opcode & 0x3f) << 2) + 4);
+        opcode = self._bytecode_array[self._index]
+        self._index += 1
+        return 'vsp = vsp - %u' % (((opcode & 0x3f) << 2) + 4)
+
+    GPRRegisterNames = ("r0", "r1", "r2", "r3", "r4", "r5", "r6", "r7",
+                        "r8", "r9", "r10", "fp", "ip", "sp", "lr", "pc")
+
+    def _PrintGPR(self, GPRMask):
+        hits = []
+        for i in range(16):
+            if GPRMask & (1 << i) != 0:
+                hits.append(self.GPRRegisterNames[i])
+        return '{%s}' % ', '.join(hits)
+
+    def _PrintRegisters(self, VFPMask, Prefix):
+        hits = []
+        for i in range(32):
+            if VFPMask & (1 << i) != 0:
+                hits.append(Prefix + str(i))
+        return '{%s}' % ', '.join(hits)
+
+    def _Decode_1000iiii_iiiiiiii(self):
+        op0 = self._bytecode_array[self._index]
+        self._index += 1
+        op1 = self._bytecode_array[self._index]
+        self._index += 1
+        #   uint16_t GPRMask = (Opcode1 << 4) | ((Opcode0 & 0x0f) << 12);
+        #   SW.startLine()
+        #     << format("0x%02X 0x%02X ; %s",
+        #               Opcode0, Opcode1, GPRMask ? "pop " : "refuse to unwind");
+        #   if (GPRMask)
+        #     PrintGPR(GPRMask);
+        GPRMask = (op1 << 4) | ((op0 & 0x0f) << 12)
+        if GPRMask == 0:
+            return 'refuse to unwind'
+        else:
+            return 'pop %s' % self._PrintGPR(GPRMask)
```

Comment:
Is implicit conversion to true preferred?
```python
if GPRMask:
    return 'pop %s' % self._PrintGPR(GPRMask)
else:
    return 'refuse to unwind'

Reply:
I think you misunderstand `C` and `Python`. 
Type of `GPRMask` is int. ` if GPRMask` always return TRUE, it's different from `if GPRMask == 0`.

Reply:
👍